File: frontend/src/types/checkin.ts
export type Mood = "good" | "okay" | "struggled";

export interface CheckIn {
  id: string;
  habit_id: string;
  progress: number;
  mood: Mood;
  note: string | null;
  ai_feedback: string | null;
  created_at: string;
}

export interface CheckInCreate {
  habit_id: string;
  progress: number;
  mood: Mood;
  note?: string;
}

File: backend/app/schemas/checkin.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class CheckInBase(BaseModel):
    progress: float
    mood: str
    note: Optional[str] = None
    ai_feedback: Optional[str] = None


class CheckInCreate(CheckInBase):
    habit_id: str


class CheckInResponse(CheckInBase):
    id: str
    habit_id: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
